### ISSUE TITLE
Transducer loss implementation length normalize?

### ISSUE THREAD

**Issue by lemonbuffer20** [speechbrain/speechbrain#875]
Hi, while using numba Transducer loss, I found that it normalizes log_p over time on both forward and backward.

https://github.com/speechbrain/speechbrain/blob/d6adc40e742107c34ae38dc63484171938b4d237/speechbrain/nnet/loss/transducer_loss.py#L106
https://github.com/speechbrain/speechbrain/blob/d6adc40e742107c34ae38dc63484171938b4d237/speechbrain/nnet/loss/transducer_loss.py#L180

I think this is different to other loss implementations; such as: warp-transducer or Numba impl. of NeMo toolkit
https://github.com/NVIDIA/NeMo/blob/c5dbf4508abaa5f54db8971b53e37266137b0399/nemo/collections/asr/parts/numba/rnnt_loss/utils/cuda_utils/gpu_rnnt_kernel.py#L164

If I comment the normalizing part, I get same result (loss value) as NeMo.
Is it a common practice? or Is there any reason to normalize?

Thanks.

**Comment by mravanelli** [comment]
Good point. I think @aheba can comment on that. On which tasks did you compare the normalized vs non-normalize loss? and what about the results?

**Comment by lemonbuffer20** [comment]
@mravanelli I have a previous training loss graph from another framework, and after changing to SpeechBrain, I wanted to check if it works well by comparing training dynamics by loss (not only WER). Because normalization above seems to only change the scale of the loss and not the gradient, the final performances were quite similar.

**Comment by mravanelli** [comment]
On a small task like TIMIT, we compared the normalized vs non-normalized version and it seems that the performance is the same. It could be interesting to try it on Librispeech as well.

**Comment by lemonbuffer20** [comment]
@mravanelli Thank you. I think we can close the issue :) We can agree that this is just how loss value is shown and not changes the gradient, thus not affects the performance.
